File: src/main/java/me/kingtux/kingserverinfo/utils/mediagui/MediaGui.java
package me.kingtux.kingserverinfo.utils.mediagui;

import java.util.ArrayList;
import java.util.HashMap;
import me.kingtux.kingserverinfo.KingServerInfoMain;
import org.bukkit.Bukkit;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;

public class MediaGui {

  private KingServerInfoMain plugin;
  private Inventory mediaGui;
  private HashMap<Items, ItemStack> guiItems = new HashMap<>();

  public MediaGui(KingServerInfoMain pl) {
    plugin = pl;
  }

  public Inventory createMediaGui() {
    mediaGui = Bukkit.createInventory(null, plugin.getConfigSettings().getMediaSize(),
        plugin.getConfigSettings().getGuiTitle());
    ArrayList<Items> allItems = plugin.getConfigSettings().getGuiItems();
    for (Items items : allItems) {
      mediaGui.setItem(items.getPosition(), items.getItem());
      guiItems.put(items, items.getItem());

    }
    return mediaGui;
  }

  public Inventory getMediaGui() {
    return mediaGui;
  }

  public HashMap<Items, ItemStack> getguiItems() {
    return guiItems;
  }
}

File: src/me/kingtux/kingserverinfo/utils/mediagui/MediaGui.kt
package me.kingtux.kingserverinfo.utils.mediagui

import me.kingtux.kingserverinfo.KingServerInfoMain
import org.bukkit.Bukkit
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack
import java.util.*

class MediaGui(private val plugin: KingServerInfoMain) {
    var mediaGui: Inventory? = null
        private set
    private val guiItems = HashMap<Items, ItemStack>()

    fun createMediaGui(): Inventory {
        mediaGui = Bukkit.createInventory(null, plugin.configSettings!!.mediaSize, plugin.configSettings!!.guiTitle)
        val allItems = plugin.configSettings!!.guiItems
        for (items in allItems!!) {
            mediaGui!!.setItem(items.position, items.item)
            guiItems[items] = items.item

        }
        return mediaGui!!
    }

    fun getguiItems(): HashMap<Items, ItemStack> {
        return guiItems
    }
}
